qiskit/optimization/algorithms/multistart_optimizer.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Defines an abstract class for multi start optimizers. A multi start optimizer is an optimizer
+that may run minimization algorithm for the several time with different initial guesses to achieve
+better results. This implementation is suitable for local optimizers."""
+
+import logging
+import time
+from abc import ABC
+from typing import Optional, Callable
+
+import numpy as np
+from scipy.stats import uniform
+
+from qiskit.optimization import QuadraticProgram, INFINITY
+from qiskit.optimization.algorithms import OptimizationAlgorithm, OptimizationResult
+
+logger = logging.getLogger(__name__)
+
+
+# we disable a warning: "Method 'a method' is abstract in class 'OptimizationAlgorithm' but
+# is not overridden (abstract-method) since this class is not intended for instantiation
+# pylint: disable=W0223
+class MultiStartOptimizer(OptimizationAlgorithm, ABC):
```

Comment:
As this class has but a single method, which does not use self at all, it perhaps could have been considered to have had this a function at the module level which a class wanting to have this capability could directly call. Maybe its useful though to have a class implement this as a type since then it can be understood by type if needed that the optimizer has this aspect. Just a comment...

Reply:
It is a good point. I have a long experience in object oriented programming and that's why I think of classes first and a plain function hanging out somewhere outside of a class still looks somehow weird to me. I don't say that I'm against them and this suggestion especially, but it seems to me as matter of taste.
So, I can move out the function to the module level but prefer to have an abstract class :) What do you think?

Reply:
If this class had a init method that took trials and clip and these were passed up in a super init call by the derived class and stored for later use in this class, rather than being stored in each derived class then the method would end up as an instance one as you have it, and it would seem more useful as a class in its own right. I don't think trials and clip are used outside of this by the derived class, right. Also setter/getters could be added here to allow these to be updated in this class, in a common way, if that seems useful/desirable.

Reply:
Totally agreed. Made `MultiStartOptimizer` more like an abstract class. I should not have hurried.